mm2src/coins/z_coin.rs

```diff
@@ -1321,6 +1322,38 @@ impl UtxoTxBroadcastOps for ZCoin {
     }
 }
 
+#[async_trait]
+#[cfg_attr(test, mockable)]
+impl ListUtxoOps for ZCoin {
```

Comment:
ZCoin is not assumed to work with transparent UTXOs. This trait should not be implemented for it.

Reply:
Unfortunately, `ZCoin::preimage_trade_fee_required_to_send_outputs` is based on `utxo_common::preimage_trade_fee_required_to_send_outputs`, that requires `ListUtxoOps` to [generate](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/coins/utxo/utxo_common.rs#L2538) a preimage transaction.
https://github.com/KomodoPlatform/atomicDEX-API/blob/778461afccf42c17da0e02394ef6890a092a7a99/mm2src/coins/z_coin.rs#L1425-L1433

So I think we need to implement this method requesting z_unspents instead and then we can get rid of the `ListUtxoOps` trait implementation for `ZCoin`.
Do you think I need to do it within this PR?

Reply:
I can refactor it during next iteration of Zcoin implementation. Not required for now.

Reply:
But I already have a different idea: if only one method from a trait is required to split these implementations, then we can maybe add a trait like `TradePreimageUtxoOps` and implement only it for `Zcoin`? I didn't dig in the code regarding this, so not sure how feasible it will be. @sergeyboyko0791

Reply:
Could you please explain more how `TradePreimageUtxoOps` can avoid implementing `ListUtxoOps` for `ZCoin`? 😅

Reply:
I need to check this in more details myself. It will require some time to perform this refactoring, so we can leave this as is for now.